backend/app/notify/telegram.go

```diff
@@ -155,30 +162,42 @@ func (t *Telegram) sendMessage(ctx context.Context, b []byte, chatID string) err
 }
 
 func buildTelegramMessage(req Request) ([]byte, error) {
-	from := req.Comment.User.Name
-	if req.Comment.ParentID != "" {
-		from += " → " + req.parent.User.Name
-	}
-	from = "*" + from + "*"
-	link := fmt.Sprintf("↦ [original comment](%s)", req.Comment.Locator.URL+uiNav+req.Comment.ID)
+	commentURLPrefix := req.Comment.Locator.URL + uiNav
+
+	msg := "New reply to comment"
 	if req.Comment.PostTitle != "" {
-		link = fmt.Sprintf("↦ [%s](%s)", escapeTitle(req.Comment.PostTitle), req.Comment.Locator.URL+uiNav+req.Comment.ID)
+		msg += fmt.Sprintf(" for %q", req.Comment.PostTitle)
+	}
+	msg += ":"
+
+	if req.Comment.ParentID != "" {
+		msg += fmt.Sprintf(
+			"\n[Original comment](%s) from %s at %s:\n%s",
+			commentURLPrefix+req.parent.ID,
+			escapeText(req.parent.User.Name),
+			escapeText(req.parent.Timestamp.Format("02.01.2006 at 15:04")),
+			escapeText(req.parent.Orig),
+		)
 	}
 
-	msg := fmt.Sprintf("%s\n\n%s\n\n%s", from, req.Comment.Orig, link)
+	msg += fmt.Sprintf(
+		"\n[Reply](%s) from %s at %s:\n%s",
+		commentURLPrefix+req.Comment.ID,
+		escapeText(req.Comment.User.Name),
+		escapeText(req.Comment.Timestamp.Format("02.01.2006 at 15:04")),
+		escapeText(req.Comment.Orig),
+	)
 	msg = html.UnescapeString(msg)
-	body := struct {
-		Text string `json:"text"`
-	}{Text: msg}
+	body := telegramMsg{Text: msg, ParseMode: "MarkdownV2"}
 	b, err := json.Marshal(body)
 	if err != nil {
 		return nil, err
 	}
 	return b, nil
 }
 
-func escapeTitle(title string) string {
-	escSymbols := []string{"[", "]", "(", ")"}
+func escapeText(title string) string {
```

Comment:
I'm not sure about this one. The original ticket for all this escaping indicated no improvement for the initial problem https://github.com/umputun/remark42/issues/839

unless you see something really helpful due to this, I suggest dropping it completely

Reply:
I saw messages not sending without escaping, status code 400. We can't drop it according to my tests with the verify messages.